File: Leetcode/4139-minimum-absolute-distance-between-mirror-pairs/solution.cpp
static const int speedup = []() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(NULL);
    return 0;
}();
class Solution {
private:
    int reverse(int x){
        int temp = 0;
        while(x>0)
        {
            temp = x%10+temp*10;
            x /= 10;
        }
        return temp;
    }
public:
    int minMirrorPairDistance(vector<int>& nums) {
        unordered_map<int, int> map;
        int n=nums.size(),ans = nums.size();
        for(int i=n-1;i>=0;i--){
            int rev = reverse(nums[i]);
            if(map.count(rev))
                ans = min(ans,map[rev]-i);
            map[nums[i]] = i;
        }
        return ans==n?-1:ans;
    }
};

File: Leetcode/4139-minimum-absolute-distance-between-mirror-pairs/solution.java
class Solution {
    public int reverse(int x){
        int temp = 0;
        while(x>0)
        {
            temp = x%10+temp*10;
            x /= 10;
        }
        return temp;
    }
    public int minMirrorPairDistance(int[] nums) {
        Map<Integer,Integer> map = new HashMap<>();
        int ans = nums.length;
        for(int i =nums.length-1;i>-1;i--){
            int rev = reverse(nums[i]);
            if(map.containsKey(rev))
                ans = Math.min(ans,map.get(rev)-i);
            map.put((nums[i]),i);
        }
        if(ans==nums.length)
            return -1;
        return ans;
    }
}
